ast/schema_test.go

```diff
@@ -191,3 +191,98 @@ func TestParseSchemaWithSchemaBadSchema(t *testing.T) {
 		t.Fatalf("Incorrect return from parseSchema with a bad schema")
 	}
 }
+
+func TestAnyOfSchema(t *testing.T) {
+
+	// Test 1 & 3: anyOf extends the core schema
+	extendCoreStaticPropsExp1 := []*types.StaticProperty{types.NewStaticProperty("AddressLine", types.S)}
+	extendCoreStaticPropsExp2 := []*types.StaticProperty{
+		types.NewStaticProperty("State", types.S),
+		types.NewStaticProperty("ZipCode", types.S)}
+	extendCoreStaticPropsExp3 := []*types.StaticProperty{
+		types.NewStaticProperty("County", types.S),
+		types.NewStaticProperty("PostCode", types.N)}
+	extendCoreSchemaTypeExp := types.Or(types.NewObject(extendCoreStaticPropsExp3, nil),
+		types.Or(
+			types.NewObject(extendCoreStaticPropsExp1, nil),
+			types.NewObject(extendCoreStaticPropsExp2, nil)))
+
+	// Test 2: anyOf is inside the core schema
```

Comment:
💭 Would it make sense to move the fixtures that are only used once into the `tests` element definitions below? It might be nicer to see most of it in one place.... Well, this is fine, too.

Reply:
Thank you for the thorough check! I might need some clarifications here, though. Do you mean avoiding assigning local variables that are only used once or moving each test section to where they're used? I'm not sure how to do the latter since the tests are inside brackets... thanks in advance!

Reply:
I meant instead of defining `anyOfParentVarType` here and `anyOfExtendCoreSchema ` in `compile_test.go`, and having a list element of
```go
		{
			note:     "anyOf as parent",
			schema:   anyOfSchemaParentVariation,
			expected: anyOfParentVarType,
		},
```
below, we could inline the definition there,

```go
		{
			note:     "anyOf as parent",
			schema:   `{
			    "$schema": "http://json-schema.org/draft-04/schema#",
			    "anyOf": [
			        {
			            "type": "object",
			            "properties": {
			                "State":   { "type": "string" },
			                "ZipCode": { "type": "string" }
			            },
			        },
			        {
			            "type": "object",
			            "properties": {
			                "County":   { "type": "string" },
			                "PostCode": { "type": "string" }
			            },
			        }
			    ]
			}`,
			expected: types.Or(
				types.NewObject([]*types.StaticProperty{
					types.NewStaticProperty("State", types.S),
					types.NewStaticProperty("ZipCode", types.S)}, nil),
				types.NewObject([]*types.StaticProperty{
					types.NewStaticProperty("County", types.S),
					types.NewStaticProperty("PostCode", types.S)}, nil),
				),
		},
```

🤔 Now, trying this out it actually _is_ quite cumbersome, too. I guess I'd be OK with either... but lets at least move the json schema constants inside of this test if they're only used here. What do you think?

Reply:
I see. They're also used in `compiled_test.go` actually to ensure that they don't return any compilation errors. The other schema constants also lie in `compiled_test.go`.

Reply:
Sorry I had overlooked that other usage!